### ISSUE TITLE
Illegal Instruction (core dumped)

### ISSUE THREAD

**Issue by adithya-subramanian** [dmlc/mxnet#5361]
I am learning mxnet but while using mod.fit the program always throws Illegal Instruction (core dumped) .The code I am using is completely taken from the mxnet.io website the problem remains same even if I change the data iterator to different function.
please help thank you

The code is:
import mxnet as mx
import numpy as np

class SimpleBatch(object):
    def __init__(self, data, label, pad=None):
        self.data = data
        self.label = label
        self.pad = pad

class SimpleIter:
    def __init__(self, mu, sigma, batch_size, num_batches):
        self.mu = mu
        self.sigma = sigma
        self.batch_size = batch_size
        self.num_batches = num_batches
        self.data_shape = (batch_size, mu.shape[1])
        self.label_shape = (batch_size, )
        self.cur_batch = 0

    def __iter__(self):
        return self

    def reset(self):
        self.cur_batch = 0        

    def __next__(self):
        return self.next()

    @property
    def provide_data(self):
        return [('data', self.data_shape)]

    @property
    def provide_label(self):
        return [('softmax_label', self.label_shape)]

    def next(self):
        if self.cur_batch < self.num_batches:
            self.cur_batch += 1
            num_classes = self.mu.shape[0]
            label = np.random.randint(0, num_classes, self.label_shape)
            data = np.zeros(self.data_shape)
            for i in range(num_classes):
                data[label==i,:] = np.random.normal(
                    self.mu[i,:], self.sigma[i,:], (sum(label==i), self.data_shape[1])) 
            return SimpleBatch(data=[mx.nd.array(data)], label=[mx.nd.array(label)], pad=0)
        else:
            raise StopIteration

class SyntheticData:
    """Genrate synthetic data
    """
    def __init__(self, num_classes, num_features):
        self.num_classes = num_classes
        self.num_features = num_features
        self.mu = np.random.rand(num_classes, num_features)
        self.sigma = np.ones((num_classes, num_features)) * 0.1

    def get_iter(self, batch_size, num_batches=10):
        return SimpleIter(self.mu, self.sigma, batch_size, num_batches)
		

# mlp
net = mx.sym.Variable('data')
net = mx.sym.FullyConnected(net, name='fc1', num_hidden=64)
net = mx.sym.Activation(net, name='relu1', act_type="relu")
net = mx.sym.FullyConnected(net, name='fc2', num_hidden=10)
net = mx.sym.SoftmaxOutput(net, name='softmax')
# synthetic 10 classes dataset with 128 dimension 
data = SyntheticData(10, 128)
mx.viz.plot_network(net)
mod = mx.mod.Module(symbol=net, 
                    context=mx.cpu(),
                    data_names=['data'], 
                    label_names=['softmax_label'])

import logging
logging.basicConfig(level=logging.INFO)

batch_size=32
mod.fit(data.get_iter(batch_size), 
        eval_data=data.get_iter(batch_size),
        optimizer='sgd',
        optimizer_params={'learning_rate':0.1},
        eval_metric='acc',
        num_epoch=5)

y = mod.predict(data.get_iter(batch_size))
'shape of predict: %s' % (y.shape,)

**Comment by piiswrong** [comment]
```
tmp.py:74: UserWarning: label_names has been deprecated. For prediction only, do not set label_names. For training, please use the new mxnet.loss.* classes for symbol
  label_names=['softmax_label'])
/home/junyuanx/mxnet/python/mxnet/module/module.py:392: UserWarning: Training with ***Output symbols is deprecated. Please use mxnet.loss.* classes instead. See mxnet.mod.Module's document for usage
  "Training with ***Output symbols is deprecated. "
INFO:root:Epoch[0] Train-accuracy=0.112500
INFO:root:Epoch[0] Time cost=0.274
INFO:root:Epoch[0] Validation-accuracy=0.087500
INFO:root:Epoch[1] Train-accuracy=0.125000
INFO:root:Epoch[1] Time cost=0.300
INFO:root:Epoch[1] Validation-accuracy=0.075000
INFO:root:Epoch[2] Train-accuracy=0.159375
INFO:root:Epoch[2] Time cost=0.284
INFO:root:Epoch[2] Validation-accuracy=0.290625
INFO:root:Epoch[3] Train-accuracy=0.215625
INFO:root:Epoch[3] Time cost=0.314
INFO:root:Epoch[3] Validation-accuracy=0.175000
INFO:root:Epoch[4] Train-accuracy=0.278125
INFO:root:Epoch[4] Time cost=0.305
INFO:root:Epoch[4] Validation-accuracy=0.100000
```
works fine for me. Try pulling latest master and rebuild?

**Comment by adithya-subramanian** [comment]
Thanks the latest build worked out
